src/vignette/util/thumbnail.clj

```diff
@@ -42,7 +48,23 @@
   [args]
   (perf/timing :imagemagick (apply sh args)))
 
-(defn original->thumbnail
+(defmulti original->thumbnail (fn [resource thumb-map]
+                                (type resource)))
+
+(defmethod original->thumbnail File [resource thumb-map]
+  (file->thumbnail resource thumb-map))
+
+(defmethod original->thumbnail LocalStoredObject [resource thumb-map]
+  (let [content-type (content-type resource)]
+    (when (contains? mime-blacklist content-type)
+      (throw+ {:type :convert-error
+               :response-code 404
+               :content-type content-type
+               :thumb-map thumb-map}
+              "unsupported content-type"))
+    (file->thumbnail (file-stream resource) thumb-map)))
```

Comment:
Why is the logic different here depending on if it's a `File` or a `LocalStorageObject`? I would have thought that may this would be a `:pre` condition or a check of some form.